spynnaker/pyNN/models/neuron/synapse_dynamics/synapse_dynamics_structural.py

```diff
@@ -0,0 +1,567 @@
+        if spec.current_region != constants.POPULATION_BASED_REGIONS. \
+                SYNAPSE_DYNAMICS.value:
+            spec.switch_write_focus(region)
+
+        if self._p_rew * 1000. < machine_time_step / 1000.:
+            # Fast rewiring
+            spec.write_value(data=1)
+            spec.write_value(
+                data=int(machine_time_step / (self._p_rew * 10 ** 6)),
+                data_type=DataType.INT32)
+        else:
+            # Slow rewiring
+            spec.write_value(data=0)
+            spec.write_value(
+                data=int((self._p_rew * 10 ** 6) / float(machine_time_step)),
+                data_type=DataType.INT32)
+
+        # scale the excitatory weight appropriately
+        spec.write_value(data=int(round(self._weight * weight_scales[0])),
+                         data_type=DataType.INT32)
+        # scale the inhibitory weight appropriately
+        spec.write_value(data=int(round(self._weight * weight_scales[1])),
+                         data_type=DataType.INT32)
+        spec.write_value(data=self._delay, data_type=DataType.INT32)
+        spec.write_value(data=int(self._s_max), data_type=DataType.INT32)
+        spec.write_value(data=int(self._lateral_inhibition),
+                         data_type=DataType.INT32)
+        spec.write_value(data=int(self._random_partner),
+                         data_type=DataType.INT32)
+        # write total number of atoms in the application vertex
+        spec.write_value(data=app_vertex.n_atoms, data_type=DataType.INT32)
+        # write local low, high and number of atoms
+        spec.write_value(data=post_slice[0], data_type=DataType.INT32)
+        spec.write_value(data=post_slice[1], data_type=DataType.INT32)
+        spec.write_value(data=post_slice[2], data_type=DataType.INT32)
+
+        # write the grid size for periodic boundary distance computation
+        spec.write_value(data=self._grid[0], data_type=DataType.INT32)
+        spec.write_value(data=self._grid[1], data_type=DataType.INT32)
+
+        # write probabilities for elimination
+        spec.write_value(data=int(self._p_elim_dep * (2 ** 32 - 1)),
+                         data_type=DataType.UINT32)
+        spec.write_value(data=int(self._p_elim_pot * (2 ** 32 - 1)),
+                         data_type=DataType.UINT32)
+
+        # write the random seed (4 words), generated randomly,
+        # but the same for all postsynaptic vertices!
+        for seed in self._seeds:
+            spec.write_value(data=seed)
+
+        # write local seed (4 words), generated randomly!
+        for _ in range(4):
+            spec.write_value(data=np.random.randint(0x7FFFFFFF))
+
+        # Write presynaptic (sub)population information
+
+        self.__write_presynaptic_information(
+            spec, application_graph, machine_graph,
+            app_vertex, post_slice, machine_vertex, graph_mapper,
+            routing_info)
+
+    def __compute_aux(self, application_graph, machine_graph,
```

Comment:
big function, think about breaking apart into smaller ones. my rule of thumb, if beond a page, break